src/json.hpp

```diff
@@ -116,6 +116,19 @@ SOFTWARE.
 #elif (defined(__cplusplus) && __cplusplus >= 201402L) || (defined(_HAS_CXX14) && _HAS_CXX14 == 1)
     #define JSON_HAS_CPP_14
 #endif
+/*
+In Visual Studio 2017 there are warnings on min
+and max in this header having not enough args.
+This is because in C min and max take args and
+well are not needed anyway so we undefine them
+first to silence these annoying warnings!!!
+*/
+#ifdef min
```

Comment:
I don't see the point of undefining a symbol that was not defined by the library. This feels like cleaning up after MSVC...

Reply:
It only happens in MSVC (1911) for some reason yes.